docs/tutorials/interceptors.md

```diff
@@ -0,0 +1,123 @@
+
+To demonstrate the flow on a more specific example, lets take two interceptors **gtp-cache** and **pii-anonymizer** configured for the **GPT-4** model:
+
+```json
+//configuration in AI DIAL Core
+    "models": {
+        "chat-gpt-4": {
+            "interceptors": ["gtp-cache", "pii-anonymizer"]
+        }
+```
+
+1. AI DIAL Core receives a request from AI DIAL Chat to query the GPT-4 model.
+2. The first interceptor, **gpt-cache**, checks the cache for the request. If found, the response is returned to AI DIAL Core; if not, the request is forwarded to **pii-anonymizer**.
+3. The **pii-anonymizer** interceptor anonymizes any personally identifiable information (PII) in the request and forwards it to **GPT-4**.
+4. After all interceptors have processed the request, AI DIAL Core sends it directly to the GPT-4 model.
+5. AI DIAL Core retrieves the response from **GPT-4** and forwards it to **pii-anonymizer**.
+6. The **pii-anonymizer** interceptor restores the original PII in the response and passes it to **gpt-cache**.
+7. The **gpt-cache** interceptor stores the response in the cache and returns it to AI DIAL Core.
+8. AI DIAL Core sends the final response back to AI DIAL Chat.
+
+## Configuration
+
+Interceptors can be defined and assigned in AI DIAL Core [dynamic settings](https://github.com/epam/ai-dial-core?tab=readme-ov-file#dynamic-settings).
+
+### Add
+
+1. Add `interceptors` section to [AI DIAL Core dynamic settings](https://github.com/epam/ai-dial-core/blob/development/sample/aidial.config.json#L3).
+2. Define a list of interceptors that will be available for your deployments: 
+
+```json
+    "interceptors": {
+        "gpt-cache": {
+            "endpoint": "${INTERCEPTOR_SERVICE_URL}/openai/deployments/gpt-cache/chat/completions",
+            "description": "description"
+        },
+        "pii-anonymizer": {
+            "endpoint": "${INTERCEPTOR_SERVICE_URL}/openai/deployments/pii-anonymizer/chat/completions",
+            "description": "description"
+        }
+    }
+    ...
+```
+
+### Assign
+
+To assign `interceptors` to `applications` and `models`:
+
+1. Add `interceptors` array to `models` and `applications`.
+2. Define interceptors from the available list: 
+
+```json
+    { 
+    "applications": {
+        "app": {
+            "interceptors": ["gpt-cache", "pii-anonymizer"]                
+        },
+        ...
+    },
+    "models": {
+        "chat-gpt-4": {
+            "interceptors": ["gpt-cache", "pii-anonymizer"]            
+        },
+        ...
+    },
+    ...
+    }
+```
+
+**Note**: make sure that chat completion interceptors are only used in chat models or application, embeddings interceptors are only used in embeddings models.
+
+## Interceptor Endpoint Specification
```

Comment:
This section is redundant since it repeats which already has been said or demonstrated above.

Reply:
fixed